src/ETL/dataset.py

```diff
@@ -105,9 +112,16 @@ def find_matching_point(
     Additional value is given to a grid coordinate that is close to the center of the tif.
     """
     start_date = datetime.strptime(start, "%Y-%m-%d")
-    tif_slope_tuples = [
-        Engineer.load_tif(p, start_date=start_date, num_timesteps=None) for p in tif_paths
-    ]
+    tif_slope_tuples = []
+    for p in tif_paths:
+        blob = bucket.blob(str(p))
+        local_path = f"{temp_dir}/{p.name}"
+        blob.download_to_filename(local_path)
+        tif_slope_tuples.append(
+            Engineer.load_tif(local_path, start_date=start_date, num_timesteps=None)
+        )
+        Path(local_path).unlink()
```

Comment:
The path being provided is now a Google Cloud URI so the file has to be downloaded first (and deleted after to save space)